.github/workflows/main_workflow.yml

```diff
@@ -253,11 +255,18 @@ jobs:
           sudo chmod +x skaffold
           sudo install skaffold /usr/local/bin/
           skaffold --help
-
+          
+          # install docker (macOS specific)
+          HOMEBREW_NO_AUTO_UPDATE=1 brew install --cask docker
+          sudo /Applications/Docker.app/Contents/MacOS/Docker --unattended --install-privileged-components
+          open -a /Applications/Docker.app --args --unattended --accept-license
+          echo "We are waiting for Docker to be up and running. It can take over 2 minutes..."
+          while ! /Applications/Docker.app/Contents/Resources/bin/docker info &>/dev/null; do sleep 1; done
```

Comment:
this is the original shell code snippet (https://github.com/actions/runner/issues/1456#issue-1042407558) to install Docker on macOS runner. 

I would like to improve the last while-loop to have a maximum number of iterations; otherwise it might loop forever.